docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,289 @@
+
+The EIP introduces one or more new opcodes to the EVM.
+
+### Test Vectors
+
+- [ ] Memory expansion
+    - [ ] Verify that the opcode execution results in the correct memory expansion, being by offset or size or interaction of both parameters (Size of zero should never result in memory expansion, regardless of offset value). Test at least the following memory expansion sizes
+        - [ ] Zero bytes expansion
+        - [ ] Single byte expansion
+        - [ ] 31 bytes expansion
+        - [ ] 32 bytes expansion
+        - [ ] 33 bytes expansion
+        - [ ] 64 or more bytes expansion
+- [ ] Stack over/underflows
+    - [ ] If the opcode pushes one or more items to the stack, and the opcode pushes more elements than it pops, verify that the opcode execution results in exeptional abort when pushing elements to the stack would result in the stack having more than 1024 elements.
+    - [ ] If the opcode pops one or more items to the stack, or it has a minimum stack height of one or more, verify that the opcode execution results in exeptional abort then stack has 1 less item than the minimum stack height expected.
+- [ ] Execution context
+    - [ ] Normal call
+    - [ ] Static call
+        - [ ] Verify exeptional abort if the opcode attempts to modify the code, storage or balance of an account
+    - [ ] Delegate call
+        - [ ] If the opcode modifies the storage of the account currently executing it, verify that only the account that is delegating execution is the one that has its storage modified.
+        - [ ] If the opcode modifies the balance of the account currently executing it, verify that only the account that is delegating execution is the one that has its balance modified.
+        - [ ] If the opcode modifies the code of the account currently executing it, verify that only the account that is delegating execution is the one that has its code modified.
+    - [ ] Code call
+    - [ ] Initcode
+        - [ ] Verify opcode behaves as expected when running during the initcode phase of contract creation
+            - [ ] Initcode of a contract creating transaction.
+            - [ ] Initcode of a contract creating opcode (including itself if opcode creates a contract).
+    - [ ] Transaction context
+        - [ ] If opcode changes behavior depending on particular transaction properties, test using multiple values for each property.
+    - [ ] Block context
+        - [ ] If opcode changes behavior depending on particular block properties, test using multiple values for each property.
+    - [ ] EOF Container Context
+        - [ ] If opcode changes behavior depending on particular EOF container properties, test using multiple values for each property.
+- [ ] Return data
+    - [ ] Verify proper return data buffer modification if the opcode is meant to interact with it, otherwise verify that the return data buffer is unnaffected
+        - [ ] At current call context.
+        - [ ] At parent call context.
+- [ ] Gas usage
```

Comment:
Most bugs in implementations arise due to "order-of-operations" gas calculations. For instance when should the 63/64 gas cap be calculated? Before or after gas deduction?